commit 6c47ad415246cc3a3edac2e1637106d330574c85
Author: Mark Hatle <mark.hatle@windriver.com>
Date:   Fri May 11 19:05:09 2012 -0500

    RpmHeader.cc: RPM 5.4.9 integration
    
    _RPMVSF_xxx - NODIGEST, NOSIGNATURES, NOHEADER, and NOPAYLOAD
    are now obsolete and no longer supported.

    Upstream-Status: Inappropriate [other]
      OE version is too old, when we uprev if this patch is still need it
      should be sent upstream at that time.
    
    Signed-off-by: Mark Hatle <mark.hatle@windriver.com>

diff --git a/zypp/target/rpm/RpmHeader.cc b/zypp/target/rpm/RpmHeader.cc
index 1224ecd..c96dae6 100644
--- a/zypp/target/rpm/RpmHeader.cc
+++ b/zypp/target/rpm/RpmHeader.cc
@@ -230,10 +230,12 @@ RpmHeader::constPtr RpmHeader::readPackage( const Pathname & path_r,
   librpmDb::globalInit();
   rpmts ts = ::rpmtsCreate();
   unsigned vsflag = RPMVSF_DEFAULT;
+#if !defined(_RPM_5)
   if ( verification_r & NODIGEST )
     vsflag |= _RPMVSF_NODIGESTS;
   if ( verification_r & NOSIGNATURE )
     vsflag |= _RPMVSF_NOSIGNATURES;
+#endif
   ::rpmtsSetVSFlags( ts, rpmVSFlags(vsflag) );
 
   Header nh = 0;
